mobileapps/views.py

```diff
@@ -296,3 +300,206 @@ def delete(self, request, mobileapp_id):
+                return Response({'message': _('Notification Provider not found')}, status.HTTP_404_NOT_FOUND)
+            notification_type = get_notification_type(u'open-edx.mobileapps.notifications')
+            notification_message = NotificationMessage(
+                namespace=mobile_app.identifier,
+                msg_type=notification_type,
+                payload={'title': title,
+                         'message': message,
+                         'send_to_all': True
+                         }
+            )
+
+            # Send the notification_msg to the Celery task
+            publish_mobile_apps_notifications_task.delay([], notification_message, api_keys, notification_provider)
+
+        except Exception, ex:  # pylint: disable=broad-except
+            return Response({'message':  _('Server error')}, status.HTTP_500_INTERNAL_SERVER_ERROR)
+
+        return Response({'message': _('Accepted')}, status.HTTP_202_ACCEPTED)
+
+
+class MobileAppSelectedUsersNotifications(SecureListCreateAPIView):
+    """
+    **Use Cases**
+
+        send a push notification to a list of given users of an app
+        using urban airship push notification backend.
+
+    **Example Requests**
+
+        POST /api/mobileapps/{id}/users/notification
+
+        The body of the POST request must include the following parameters.
+
+        * title: notification title
+        * message: notification message
+        * users: comma separated list of user ids
+
+    **Response Values**
+
+        If the request is successful, the request returns an HTTP 202 "Accepted" response.
+
+        The HTTP 202 response has the following value.
+
+        * message: Accepted
+    """
+
+    def post(self, request, pk):
+
+        message = request.data.get('message', None)
+        if not message:
+            return Response({'message': _('message is missing')}, status.HTTP_400_BAD_REQUEST)
+        title = request.data.get('title', None)
+        if not title:
+            return Response({'message': _('title is missing')}, status.HTTP_400_BAD_REQUEST)
+        user_ids = request.data.get('users', None)
+        if not user_ids:
+            return Response({'message': _('Users list is empty')}, status.HTTP_400_BAD_REQUEST)
+
+        try:
+            mobile_app = MobileApp.objects.get(pk=pk)
+            if not mobile_app.is_active:
+                return Response({'message': _('Mobile app is inactive')}, status.HTTP_403_FORBIDDEN)
+        except ObjectDoesNotExist:
+            return Response({'message': _('Mobile app does not exist')}, status.HTTP_404_NOT_FOUND)
+
+        try:
+            api_keys = get_api_keys(mobile_app)
+            notification_provider = get_provider_name(mobile_app)
+            if not notification_provider:
+                return Response({'message': _('Notification Provider not found')}, status.HTTP_404_NOT_FOUND)
+            notification_type = get_notification_type(u'open-edx.mobileapps.notifications')
```

Comment:
we should make a new method which should make a push notification message with `title` and `app_identifier` params and should reuse that method all over.